### ISSUE TITLE
Feature request: Support for Uploadcare CDN API

### ISSUE THREAD

**Issue by maoosi** [ascorbic/unpic#18]
Very nice library! Would love to see support for [Uploadcare](https://uploadcare.com), which is a great image upload and delivery platform.

Image processing docs: https://uploadcare.com/docs/transformations/image/
Full API reference for images delivery: https://uploadcare.com/api-refs/url-api/

**Comment by davedbase** [comment]
I use Uploadcare heavily on a project and will likely be contributing this CDN adapter in a few weeks. My only concern is that it has support for a lot of extra transform modifiers. I might be incorrect but my understanding is that Unpic doesn't support a modifiers-like pattern, so the only benefit we get here is the basic scaling functionality.

Perhaps Unpic should consider a modifiers approach so that we can support every provider's unique abilities. ie. filters, effects, etc.

**Comment by ascorbic** [comment]
Hi. I need to explain it better, but the general idea is that it preserves any CDN-specific modifiers that are already in the URL, while allowing the other params to be modified. There could be something to be said for allowing more control over individual CDN params, but it's not in scope for now

**Comment by danestves** [comment]
I think I can work on this adapter, only if you are open to make some transformations of the URL on the adapter itself @ascorbic 

Since Uploadcare doesn't use query params and instead use URL paths to make that, I want to know your opinion

**Comment by ascorbic** [comment]
Yes, that would be fine. If I'm understanding what you mean, there are several CDNs that work like that. See the Cloudinary transformer for an example, or IPX which is a simpler one.
